### PR TITLE
Issue 328: Content no longer loads

### LINKED ISSUE #328: Content no longer loads
**Describe the bug**
See Olney email also, now it doesn't work for me. screenshot below. It does not get to the feedback settings or reading question...

**To Reproduce**
Use my test content with the test login

**Expected behavior**
Normal continuation of unit

**Screenshots**
mofacts_app and 14 more pages - Personal - Microsoft​ Edge 10_8_2021 2_56_57 PM

### PR DESCRIPTION
**Pull request by MegaGeese**
Added a more robust check for if stimDisplayTypeMap is undefined or has undefined parameters
Fixed bug with instructions not loading.

tested locally with stim and tdf files in use in the issue and content loaded. 
closes #328

### CODE REVIEW

File: mofacts/client/views/experiment/card.js [hunk 1]
```diff
@@ -876,13 +876,15 @@ function checkUserAudioConfigCompatability(){
 function curStimHasSoundDisplayType() {
   const currentStimuliSetId = Session.get('currentStimuliSetId');
   const stimDisplayTypeMap = Session.get('stimDisplayTypeMap');
-  return currentStimuliSetId && stimDisplayTypeMap ? stimDisplayTypeMap[currentStimuliSetId].hasAudio : false;
+  const audioEnabled = stimDisplayTypeMap && stimDisplayTypeMap[currentStimuliSetId] ? stimDisplayTypeMap[currentStimuliSetId].hasAudio : false;
```
Comment (andrewtackett) [inline mofacts/client/views/experiment/card.js:879]: stimDisplayTypeMap *should* always be defined.  Why was it undefined?

Comment (MegaGeese) [reply mofacts/client/views/experiment/card.js:879]: The error message reported in issue 328 said "Cannot read property of undefined (reading 'has audio')

Running the query in getStimDisplayTypeMap on the staging db and got a stimulus set id for a different tdf than the one that was throwing the error. So the error is caused at or before the upsertStimFile function. However I wasnt able to reproduce this error locally using the same tdf and stim as in the intial issue. 

Basically it was undefined because the stim file wasnt being posted to the db in upsertStimFile

Comment (andrewtackett) [reply mofacts/client/views/experiment/card.js:879]: "Basically it was undefined because the stim file wasnt being posted to the db in upsertStimFile"

I'm unclear how this fixes that. Either we need to rewrite upsertStimFile or there is an environmental context issue.

Also we specifically don't want code like the submitted because that will cause subtle failures and allow operation to continue.  We want operation to halt if things aren't working the way we expected because otherwise we may be generating useless data and wasting everyone's time.  This is something I realized myself earlier in my career as a dev, exceptions are often wanted rather than just trying to catch them all or avoid them all because they point to a mismatch between how the problem was modeled vs reality and in many cases we want to go back and reexamine why there was a mismatch.

I'm willing to bet in this case the user data wasn't cleared out and so the tdf/stimulisetid num is from an older run but you should try to verify that or else figure out why there was an issue.
